### PR TITLE
Blog app - Processing data models

### PR DESCRIPTION
**Pull request by ArthurGC**
# Documentation: Blog app - Ruby on Rails

## :blue_book: Steps

- [x] User model: 
- I added a method that returns the 3 most recent posts for a given user.
- [x] Post model:
- I added a method that updates the posts counter for a user.
- I added a method that returns the 5 most recent comments for a given post.
- [x] Comment model:
- I added a method that updates the comments counter for a post.
- [x] Like model:
- I added a method that updates the likes counter for a post.
- [x] ApplicationController
- I added a method called current_user method that returns the first user from a database.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,5 +1,9 @@
 class User < ApplicationRecord
-  has_many :posts
-  has_many :comments
-  has_many :likes
+  has_many :posts, dependent: :destroy
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
```
Comment (arslanbisharat) [inline app/models/user.rb]: - [x]  This is a nice approach but it's recommended to use `foreign_key: 'author_id'`

File: app/models/post.rb [hunk 2]
```diff
@@ -1,5 +1,14 @@
 class Post < ApplicationRecord
   belongs_to :user
-  has_many :comments
-  has_many :likes
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
```
Comment (arslanbisharat) [inline app/models/post.rb:4]: This is a nice approach but it's recommended to use foreign_key: 'author_id'

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app/models/user.rb (hunk 1 above)
Comment (de-scholar) [reply app/models/user.rb]: :+1: This comment is addressed successfully :heavy_check_mark:

File: app/models/post.rb (hunk 2 above)
Comment (de-scholar) [reply app/models/post.rb:4]: :confused:

**Review by de-scholar — APPROVED** [review]
Hi @ArthurGC , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**